Client/WriteApiAsync.cs

```diff
@@ -380,7 +379,7 @@ await WriteData(options.OrganizationId, options.Bucket, grouped.Key, groupedPoin
         public Task WriteMeasurementsAsync<TM>(TM[] measurements, WritePrecision precision = WritePrecision.Ns,
             string bucket = null, string org = null, CancellationToken cancellationToken = default)
         {
-            return WriteMeasurementsAsync(measurements.ToList(), precision, bucket, org, cancellationToken);
+            return WriteMeasurementsAsync(measurements, precision, bucket, org, cancellationToken);
```

Comment:
This causes infinity recursive, please change to:
```suggestion
            return WriteMeasurementsAsync((IEnumerable<TM>)measurements, precision, bucket, org, cancellationToken);
```